common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -287,7 +287,7 @@ public void update(MLCreateConnectorInput updateContent, Function<String, String
             this.protocol = updateContent.getProtocol();
         }
         if (updateContent.getParameters() != null && updateContent.getParameters().size() > 0) {
-            this.parameters = updateContent.getParameters();
+            getParameters().putAll(updateContent.getParameters());
```

Comment:
Add UT and integ test? Seems this logic is not tested, otherwise the change of the behavior should break existing tests

Reply:
BTW, thanks for the fix

Reply:
Will do in the next PR.